### PR TITLE
Make racket-browse-url functions compatible with eww

### PR DESCRIPTION
**Pull request by Fackelmann**
Emac's `eww` browser opens urls such as `/path/to/a/file.html` as `https://path/to/a/file.html`. This causes the documentation functions not to open the correct documentation page (such address doesn't exist).

Instead, the url provided to `eww` must be in the form `file:///path/to/a/file.html`. I have manually tested that this format is also compatible with `browse-url-default-macosx-browser`.

This PR modifies the argument in the `browse-url` call in `racket-browse-url-using-temporary-file` to use `url` (which already has the correct format) instead of `file`.

### CODE REVIEW

**Comment by greghendershott** [comment]
Thanks for the PR!

This seems good! Before merging, I just want to find time in the next day or two to reload my brain with the history of #476 (which was the motivation for adding the code modified by the PR) to make sure.

**Comment by greghendershott** [comment]
Merged -- thanks!

**Comment by greghendershott** [comment]
@Fackelmann I'm sorry, I messed up here. I said I needed to reload my brain --- but then didn't before merging it.

My fault, not yours.

I just un-merged this.

Why: Effectively, this changed `racket-browse-url-using-temporary-file` from doing what it says, into being just `browse-url`. This breaks the default case of using an external web browser on all operating systems.

Its doc string:

```elisp
(defun racket-browse-url-using-temporary-file (url &rest _args)
  "Browse a URL via a temporary HTML file using a meta redirect.

A suitable value for the variable `racket-browse-url-function'.

Racket documentation URLs depend on anchors -- the portion of the
URL after the # character -- to jump to a location within a page.
Unfortunately on some operating systems and/or versions of Emacs,
the default handling for browsing file URLs ignores anchors. This
function attempts to avoid the problem by using a temporary HTML
file with a meta redirect as a \"trampoline\".

Although the intent is to provide a default that \"just works\",
you do not need to use this. You can customize the variable
`racket-browse-url-function' instead to be `browse-url', or
`browse-url-browser-function' in case have have customized that,
or indeed whatever you want."
```

TL;DR: Awhile ago I added a customization variable, `racket-browse-url-function`. I think you want to change its value from `racket-browse-url-using-temporary-file` to just `browse-url`. Does that work for you, using eww?

**Comment by greghendershott** [comment]
p.s. If setting `racket-browse-url-function` to `browse-url` doesn't quite work for you, then I'm happy to help you figure out what does work.

p.p.s. If not simply `browse-url`, maybe such a function (named `racket-browse-url-using-eww` ?) should even be added to Racket Mode as a convenience? Or maybe it's not a common enough use case? Not sure. Anyway, happy to discuss.

**Comment by capfredf** [comment]
@greghendershott I think my PR should be able to make `eww` at least open the file

**Comment by greghendershott** [comment]
@capfredf Oh great; I'll look at your PR #542.

**Comment by Fackelmann** [comment]
Ah, my bad. I should have read better the docstring; I apologize for the confusion. I took a look at @capfredf PR and yes, that should fix this issue as well.
